### DISCUSSION TITLE
Feature Request: Cast UUID to BLOB

### DISCUSSION THREAD

**Discussion by dsmilkov** [duckdb/duckdb#5705] [Ideas]
Casting UUID to blob is currently not supported. We need this to align between parquet files (with row-level UUIDs) generated by duckdb and parquet files generated outside duckdb. We want duckdb to return a `Buffer` in node.js (instead of `string`) when reading the UUID column, regardless if the column was generated using manual python or sql via `uuid()`.

To solve this we need to cast uuid to blob: `select uuid()::BLOB;`

We currently get:
`Error: Conversion Error: Unimplemented type for cast (UUID -> BLOB)`

Casting should hopefully be trivial since UUID is backed by a 16 byte blob.

**Comment by nsthorat** [comment]
Bumping this, we're working with bytes in a column in python, and when I use to_df() I get a pd.Series column with dtype=np.object instead of dtype=bytes on the other side. This means I have to do some extra work to convert it back to a dtype=bytes.

**Reply by Tishj** [reply]
That doesn't sound very related to this, but feel free to raise a bug report for it, sounds like we're missing something in our conversion of bytes to dataframe then 👍

**Comment by rgwood** [comment]
This would be great. I would like to go the other way too (from 16-byte blob to UUID).

**Comment by dioptre** [comment]
I added code for this here https://github.com/duckdb/duckdb/pull/18027
